pkg/generators/api_linter.go

```diff
@@ -136,6 +136,7 @@ type apiLinter struct {
 func newAPILinter() *apiLinter {
 	return &apiLinter{
 		rules: []APIRule{
+			&rules.NamingConvention{},
```

Comment:
Should we be more specific that this is only referring to ListMeta naming convention?

Reply:
I asked on Slack whether there should be other conventions that this should enforce:
https://kubernetes.slack.com/archives/C0409NGC1TK/p1724256359105049?thread_ts=1723854248.858719&cid=C0409NGC1TK

If there aren't then renaming makes sense.

Reply:
It's also for ObjectMeta. That wasn't in my original commit, but I've included it in the revised one.